src/components/CommentItem/CommentItem.tsx

```diff
@@ -0,0 +1,47 @@
+import { useContext } from 'react';
+import { Comment } from '../../types/Comment';
+import * as commentAPI from '../../api/commentAPI';
+import { PostContext } from '../../context/PostContext';
+
+type Props = {
+  comment: Comment;
+};
+
+export const CommentItem: React.FC<Props> = ({ comment }) => {
+  const { setComments, setErrorMessage } = useContext(PostContext);
+
+  const deleteComment = () => {
+    setComments(currentComments => currentComments
+      .filter(currentComment => currentComment.id !== comment.id));
+
+    commentAPI.removeComment(comment.id)
+      .catch(() => setErrorMessage('Something went wrong'));
```

Comment:
why do you filter your comments in local and then remove? what if your api will throw error and not remove item, but you already remove it locally and after page reload user will see this item again